model/flow/account.go

```diff
@@ -76,6 +78,120 @@ func (a *AccountPublicKey) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (a AccountPublicKey) CompactEncodeWithDefaultVersion() ([]byte, error) {
+	return a.compactEncodeV0()
+}
+
+func (a AccountPublicKey) CompactEncode(version uint8) ([]byte, error) {
+	switch version {
+	case 0:
+		return a.compactEncodeV0()
+	default:
+		return nil, fmt.Errorf("cannot encode account public key: encoding version %d is not supported", version)
+	}
+}
+
+func (a *AccountPublicKey) CompactDecode(data []byte) error {
+	// read version and redirect
+	// first 5 bits are encoding version
+	b := uint8(data[0]) & byte(248)  // first 5 bits are encoding version
+	encodingVersion := uint8(b) >> 3 // (mask of 11111000 and left shift 3)
+
+	switch encodingVersion {
+	case 0:
+		return a.compactDecodeV0(data)
+	default:
+		return fmt.Errorf("cannot decode account public key: encoding version %d is not supported", encodingVersion)
+	}
+}
+
+func (a *AccountPublicKey) compactDecodeV0(data []byte) error {
+
+	dataSize := len(data)
+	minDataSize := 15
+	// we need at least minDataSize bytes
+	if dataSize < minDataSize {
+		return fmt.Errorf("cannot compact decode public key: data size is too small to be decodable (expected: at least %d, got: %d)", minDataSize, dataSize)
+	}
+
+	// first 5 bits were encoding version so we have already processed them
+
+	// next bit is revoked flag
+	a.Revoked = (data[0] & byte(4)) > 0 // (mask of 00000100 and equality check (might optimize in the future))
+
+	// the last 2 remaining bits of first byte are the most significant bits of key weight
+	a.Weight = int(uint16(data[0]&byte(3))<<8 + uint16(data[1]))
```

Comment:
Concatenating with bit wise operations is faster than arithmetic ones. 
```suggestion
	a.Weight = int(uint16(data[0]&byte(3))<<8 | uint16(data[1]))
```